### ISSUE TITLE
RSPM: Custom repo configurations do not take effect.

### ISSUE THREAD

**Issue by fereshtehRS** [rstudio/rstudio#2864]
### System details

    RStudio Edition : Server
    RStudio Version : 1.2.665
    OS Version      : CentOS 7
    R Version       : 3.4.2

### Steps to reproduce the problem
- Access Tools -> Global Options -> Packages and make sure `Primary CRAN repo` is set to default `Global (CDN) - RStudio`
- Edit `/etc/rstudio/rsession.conf` configuration file and add the following:

`r-cran-repos=https://<your-package-manager-server-address>/<repo-name>/latest`  (pointing to a valid RSPM host and repo)
- Restart RStudio Server
- Optional: Restart R session in the browser
- Access Tools -> Global Options -> Packages. You still see the default setting.

### Describe the problem in detail
The settings in the configuration file `rsession` or the optional file `repos.conf` does not take effect.

### Describe the behavior you expected

The custom configuration setting should have taken effect and reflected in the UI dialog.

**Comment by javierluraschi** [comment]
@fereshtehRS mentioned this wasn't working for CentOS and Suse, just tried Suse 12 over EC2 following: [install-rstudio-server-suse-rmd](https://gist.github.com/javierluraschi/a52a89553e821683f95416907e415a5f#file-install-rstudio-server-suse-rmd).

and setting `/etc/rstudio/rsession.conf`:

```
r-cran-repos=http://<name>.<apackagemanager>.com/bleedingEdgeRepo/latest
```

got this working properly after changing `rsession.conf` and restarting the session:

<img width="1440" alt="screen shot 2018-05-29 at 2 06 27 pm" src="https://user-images.githubusercontent.com/3478847/40685589-88417aca-6349-11e8-8706-95531d6c9577.png">

**Comment by javierluraschi** [comment]
I was able to reproduce this as well in RStudio 1.1, I know what is going on, but a fix is still needed.

The behavior of `r-cran-repos` in RStudio 1.1 is to override this setting with user preferences, so, in 1.1 we can do:

```
# rsession.conf
r-cran-repos=foo
```

Then in RStudio Server:
- Restart r session
- Run `getOption("repos")`, notice the correct value.
- Tools -> Global Options - >OK
- Run `getOption("repos")`, notice the incorrect value.

Another repro:

- Close the Terminal, make sure no tab is visible
- Restart r session
- Run `getOption("repos")`, notice the correct value.
- Create a new Terminal
- Run `getOption("repos")`, notice the incorrect value.

The intermittent repro from this bug is caused by the terminal saving user preferences and by `r-cran-repos` being overridden by any user preference change.

We should change past behavior to avoid overriding `r-cran-repos` when preferences don't change.

**Comment by dfalty** [comment]
Verified fixed 1.2.1321-2
